harper-core/src/linting/closed_compounds.rs

```diff
@@ -0,0 +1,342 @@
+use crate::{
+    patterns::{ExactPhrase, Pattern},
+    Token, TokenStringExt,
+};
+
+use super::{Lint, LintKind, PatternLinter, Suggestion};
+
+macro_rules! create_closed_compound_linter {
+    ($name:ident, $phrase:literal, $correct:expr) => {
+        pub struct $name {
+            pattern: Box<dyn Pattern>,
+        }
+
+        impl Default for $name {
+            fn default() -> Self {
+                let pattern = ExactPhrase::from_phrase($phrase);
+
+                Self {
+                    pattern: Box::new(pattern),
+                }
+            }
+        }
+
+        impl PatternLinter for $name {
+            fn pattern(&self) -> &dyn Pattern {
+                self.pattern.as_ref()
+            }
+
+            fn match_to_lint(&self, matched_tokens: &[Token], source: &[char]) -> Lint {
+                let span = matched_tokens.span().unwrap();
+                let orig_chars = span.get_content(source);
+
+                Lint {
+                    span,
+                    lint_kind: LintKind::WordChoice,
+                    suggestions: vec![Suggestion::replace_with_match_case(
+                        $correct.chars().collect(),
+                        orig_chars,
+                    )],
+                    message: format!("Did you mean the closed compound `{}`?", $correct),
+                    ..Default::default()
+                }
+            }
+
+            fn description(&self) -> &'static str {
+                concat!(
+                    "Looks for incorrect spacing inside the closed compound `",
+                    $correct,
+                    "`."
+                )
+            }
+        }
+    };
+}
+
+create_closed_compound_linter!(Itself, "it self", "itself");
+create_closed_compound_linter!(Tonight, "to night", "tonight");
+create_closed_compound_linter!(Myself, "my self", "myself");
+create_closed_compound_linter!(Therefore, "there fore", "therefore");
+create_closed_compound_linter!(Misunderstand, "miss understand", "misunderstand");
+create_closed_compound_linter!(Misunderstood, "miss understood", "misunderstood");
+create_closed_compound_linter!(Misuse, "miss use", "misuse");
+create_closed_compound_linter!(Misused, "miss used", "misused");
+create_closed_compound_linter!(Postpone, "post pone", "postpone");
+create_closed_compound_linter!(Worldwide, "world wide", "worldwide");
+create_closed_compound_linter!(Overall, "over all", "overall");
+create_closed_compound_linter!(Likewise, "like wise", "likewise");
```

Comment:
This is not always an error, "like wise men", "like wise old owls", etc. The browser can distinguish: 
<img width="284" alt="image" src="https://github.com/user-attachments/assets/46a289ee-befb-491c-bf24-7e38bdd34213" />